main.tf

```diff
@@ -89,7 +90,11 @@ module "google_mysql_db" {
     retained_backups               = null
     retention_unit                 = null
   }
-
+  insights_config = {
+    query_string_length     = var.insights_config.query_string_length
+    record_application_tags = var.insights_config.record_application_tags
+    record_client_address   = var.insights_config.record_client_address
+  }
```

Comment:
Would suggest to just use `insights_config = var.insights_config`

Whatever keys need to be passed here will be flexible from the variable itself. Much like what was done here https://github.com/airasia/terraform-google-postgres_db/pull/6 for our postgres_db module

Reply:
done